nipype/interfaces/base/specs.py

```diff
@@ -316,6 +318,32 @@ def _get_sorteddict(self,
     def __all__(self):
         return self.copyable_trait_names()
 
+    def __getstate__(self):
+        """
+        Override __getstate__ so that OutputMultiObjects are correctly pickled.
+
+        >>> class OutputSpec(TraitedSpec):
+        ...     out = OutputMultiObject(traits.List(traits.Int))
+        >>> spec = OutputSpec()
+        >>> spec.out = [[4]]
+        >>> spec.out
+        [4]
+
+        >>> spec.__getstate__()['out']
+        [[4]]
+
+        >>> spec.__setstate__(spec.__getstate__())
+        >>> spec.out
+        [4]
+
+        """
+        state = super(BaseTraitedSpec, self).__getstate__()
+        for key in self.__all__:
+            _trait_spec = self.trait(key)
+            if _trait_spec.is_trait_type((OutputMultiObject, OutputMultiPath)):
```

Comment:
Aren't these the same thing?

```suggestion
            if _trait_spec.is_trait_type(OutputMultiObject):
```

Reply:
https://github.com/nipy/nipype/blob/7262b2425943e0b777b94066975b6606a2303020/nipype/interfaces/base/traits_extension.py#L465

Reply:
Yes, but I'm unsure the test will come positive either way. I preferred to be explicit (although I don't mind checking)

Reply:
Okay:

```Python
>>> import nipype.interfaces.base as nib
>>> class OutputSpec(nib.TraitedSpec):
...     out = nib.OutputMultiObject(nib.traits.Int)
>>> spec = OutputSpec()
>>> spec.trait('out').is_trait_type(nib.OutputMultiPath)
True
```